### PR TITLE
use custom ora instance

### PR DESCRIPTION
**Pull request by EvanBacon**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary.
- [ ] I've tagged the changelog entry with `[EAS BUILD API]` if it's a part of a breaking change to EAS Build API (only possible when updating `@expo/eas-build-job` package).

# Why

Prevent debug logs from breaking spinners

## Before

Screen Shot 2021-08-24 at 2 00 12 PM

## After

Screen Shot 2021-08-24 at 1 59 59 PM

# How

override logs and pause ora before logging to the console, also disable ora when EXPO_DEBUG is enabled.

# Test Plan

- `EXPO_DEBUG=1 eas build` -- static spinners
- `EXPO_APP_STORE_DEBUG=1 eas build` -- no broken logs

### CODE REVIEW

File: packages/eas-cli/src/ora.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+/**
+ * A custom ora spinner that sends the stream to stdout in CI, or non-TTY, instead of stderr (the default).
+ *
+ * @param options
+ * @returns
+ */
+export function ora(options?: oraReal.Options | string): oraReal.Ora {
+  const inputOptions = typeof options === 'string' ? { text: options } : options || {};
+  const disabled = Log.isDebug || !process.stdin.isTTY;
+  const ora = oraReal({
```
Comment (dsokal) [inline packages/eas-cli/src/ora.ts:14]: You're shadowing the function, pick a different name for the variable.

File: packages/eas-cli/src/ora.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import Log from './log';
+
+/**
+ * A custom ora spinner that sends the stream to stdout in CI, or non-TTY, instead of stderr (the default).
+ *
+ * @param options
+ * @returns
+ */
+export function ora(options?: oraReal.Options | string): oraReal.Ora {
+  const inputOptions = typeof options === 'string' ? { text: options } : options || {};
```
Comment (dsokal) [inline packages/eas-cli/src/ora.ts:12]: - Add parentheses around `options || {}`. It's helpful for readers so they don't need to think about which operator has higher precedence. 
- Prefer `?? {}` over `|| {}`.

File: packages/eas-cli/src/ora.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  const errorReal = console.error;
+
+  const oraStop = ora.stop.bind(ora);
+
+  const origStopAndPersist = ora.stopAndPersist.bind(ora);
+
+  const logWrap = (method: any, args: any[]) => {
+    oraStop();
+    method(...args);
+    ora!.start();
```
Comment (dsokal) [inline packages/eas-cli/src/ora.ts:38]:
```suggestion
    ora.start();
```
I checked out your branch locally and verified that `!` is not necessary.

File: packages/eas-cli/src/ora.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    // eslint-disable-next-line no-console
+    console.warn = warnReal;
+    // eslint-disable-next-line no-console
+    console.error = errorReal;
+  };
+
+  ora.stopAndPersist = (): oraReal.Ora => {
+    origStopAndPersist();
+    resetNativeLogs();
+    return ora!;
```
Comment (dsokal) [inline packages/eas-cli/src/ora.ts:64]:
```suggestion
    return ora;
```
Same here

File: packages/eas-cli/src/ora.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  };
+
+  ora.stop = (): oraReal.Ora => {
+    oraStop();
+    resetNativeLogs();
+    return ora!;
```
Comment (dsokal) [inline packages/eas-cli/src/ora.ts:70]:
```suggestion
    return ora;
```
And here.

File: packages/eas-cli/src/ora.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  };
+
+  // eslint-disable-next-line no-console
+  console.log = (...args: any) => logWrap(logReal, args);
+  // eslint-disable-next-line no-console
+  console.info = (...args: any) => logWrap(infoReal, args);
+  // eslint-disable-next-line no-console
+  console.warn = (...args: any) => logWrap(warnReal, args);
+  // eslint-disable-next-line no-console
+  console.error = (...args: any) => logWrap(errorReal, args);
```
Comment (dsokal) [inline packages/eas-cli/src/ora.ts:48]: This block should go to `ora.start`. You can create an ora instance first and start it later.

File: packages/eas-cli/src/ora.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  // eslint-disable-next-line no-console
+  const infoReal = console.info;
+  // eslint-disable-next-line no-console
+  const warnReal = console.warn;
+  // eslint-disable-next-line no-console
+  const errorReal = console.error;
+
+  const oraStop = ora.stop.bind(ora);
+
+  const origStopAndPersist = ora.stopAndPersist.bind(ora);
```
Comment (dsokal) [inline packages/eas-cli/src/ora.ts:33]: Use consistent naming. E.g. `oraStop` + `oraStopAndPersist` or `origStop` + `orig StopAndPersist`.

**Review by dsokal — CHANGES_REQUESTED** [review]

**Comment by dsokal** [comment]
@EvanBacon There's one more comment before we can merge it. Let me know if you want me to take over this PR.

**Review by dsokal — APPROVED** [review]

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**